third_party/gtest/CMakeLists.txt

```diff
@@ -1,21 +1,27 @@
-message(STATUS "Configuring and building gtest")
-if(MSVC)
-    # Force using shared runtime library when MSVC builds shared libraries
-    ginkgo_load_git_package(gtest_external
-        "https://github.com/google/googletest.git"
-        "6a7ed316a5cdc07b6d26362c90770787513822d4"
-        # Work around the linking errors when compiling gtest with CUDA
-        "-Dgtest_disable_pthreads=ON" "-Dgtest_force_shared_crt=ON")
-else()
-    ginkgo_load_git_package(gtest_external
-        "https://github.com/google/googletest.git"
-        "6a7ed316a5cdc07b6d26362c90770787513822d4"
-        # Work around the linking errors when compiling gtest with CUDA
-        "-Dgtest_disable_pthreads=ON"
-        "-DCMAKE_CXX_FLAGS=-fPIC")
+if(NOT GTest_FOUND)
+    message(STATUS "Fetching external GTest")
+    include(FetchContent)
+    FetchContent_Declare(
+        googletest
+        GIT_REPOSITORY https://github.com/google/googletest.git
+        GIT_TAG        6a7ed316a5cdc07b6d26362c90770787513822d4
+    )
+    # need to set the variables in CACHE due to CMP0077
+    set(gtest_disable_pthreads ON CACHE INTERNAL "")
+    if(MSVC)
+        set(gtest_force_shared_crt ON CACHE INTERNAL "")
+    endif()
+    FetchContent_GetProperties(googletest)
+    if(NOT googletest_POPULATED)
+        FetchContent_Populate(googletest)
+    endif()
+    add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR} EXCLUDE_FROM_ALL)
```

Comment:
Shouldn't the `add_subdirectory` be done only if it's not populated already? Otherwise that would be available already no? See example here:
https://cmake.org/cmake/help/latest/module/FetchContent.html#overview

Reply:
This is just me misunderstanding the purpose of *_POPULATED
I thought it meant "has this code already been downloaded in a previous CMake run" while it means "has this code already been downloaded previously in the same CMake run", i.e. if multiple CMake files load the same package. I will move to the documented version.